File: 프로그래머스/0/120825. 문자 반복 출력하기/문자 반복 출력하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
char* solution(const char* my_string, int n) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    char* answer = (char*)malloc(sizeof(char) * (strlen(my_string) * n + 1));
    
    int h = 0;
    for(int i=0; i<strlen(my_string); i++) {
        for(int k=0; k<n; k++) {
            answer[h++] = my_string[i];   
        }
    }
    answer[strlen(my_string)*n] = NULL;
    return answer;
}

File: 프로그래머스/0/120825. 문자 반복 출력하기/문자 반복 출력하기.java
class Solution {
    public String solution(String my_string, int n) {
        String answer = "";
        
        for(int i=0; i<my_string.length(); i++) {
            for(int j=0; j<n; j++) {
                answer += my_string.charAt(i);
            }
        }
        
        return answer;
    }
}

/*
class Solution {
    public String solution(String my_string, int n) {
        String answer = "";
        String[] str = my_string.split("");     // 문자열 한글자씩 분리 후 배열 저장
        
        for(int i=0; i<my_string.length(); i++) {
            answer += str[i].repeat(n);         // n번 반복하여 저장
        }
        
        return answer;
    }
}
*/